core/libraries/shortcodes/EE_Venue_Shortcodes.lib.php

```diff
@@ -143,23 +155,24 @@ protected function _parser($shortcode)
                 return $this->_venue('formatted_address');
                 break;
 
-            case '[GOOGLE_MAP_URL]':
-                return $this->_venue('gmap_url');
-                break;
-
-            case '[GOOGLE_MAP_LINK]':
-                return $this->_venue('gmap_link');
-                break;
-
-            case '[GOOGLE_MAP_IMAGE]':
-                return $this->_venue('gmap_link_img');
-                break;
-
             case '[VENUE_DETAILS_URL]':
                 return $this->_venue('permalink');
                 break;
 
         }
+        
+        if ( preg_match('/^\[GOOGLE_MAP_(LINK|IMAGE|URL)(_)?/', $shortcode, $matches) ) {
```

Comment:
actually, @joshfeck remarked we'd rather have a simpler switch statement or if statement instead of a regex comparison. 
This regex isn't very future friendly. If we want to add a new attribute to GOOGLE_MAP_LINK someday, we’ll have to pull that apart